packages/zcli-apps/tests/functional/package.test.ts

```diff
@@ -36,3 +39,53 @@ describe('package', function () {
     .catch(err => expect(err.message).to.contain('Error: invalid location'))
     .it('should display error message if package fails to create')
 })
+
+describe('zcliignore', function () {
+  const appPath = path.join(__dirname, 'mocks/single_product_ignore')
+  const tmpPath = path.join(appPath, 'tmp')
+
+  const file = readline.createInterface({
+    input: fs.createReadStream(path.join(appPath, '.zcliignore')),
+    output: process.stdout,
+    terminal: false
+  })
+
+  const ignoreArr: string[] = [] // array that holds each line of the .ignore file
+
+  file.on('line', (line) => {
+    ignoreArr.push(line) // add to array dynamically
+  })
+
+  after(async () => {
+    fs.readdir(tmpPath, (err, files) => {
+      if (err) throw err
+
+      for (const file of files) {
+        fs.unlink(path.join(tmpPath, file), (err) => {
+          if (err) throw err
+        })
+      }
+    })
+  })
+
+  test
+    .env({
+      ZENDESK_SUBDOMAIN: 'z3ntest',
+      ZENDESK_EMAIL: 'admin@z3ntest.com',
+      ZENDESK_PASSWORD: '123456' // the universal password
+    })
+    .nock('https://z3ntest.zendesk.com', api => {
+      api
+        .post('/api/v2/apps/validate')
+        .reply(200)
+    })
+    .stdout()
+    .command(['apps:package', appPath])
+    .it('should not include certain files as specified in .zcliignore', async () => {
+      const packagePath = path.join(tmpPath, (fs.readdirSync(tmpPath).filter(fn => fn.startsWith('app')) + ''))
```

Comment:
Why add `''` at the end? `fn.startsWith('app')) + ''`

Reply:
Changed from .filter to .find to make this better - I was assuming before that .filter would only return one result, as .after() cleared the tmp folder after each run. Now .find will return only the most recent package name and then this will be turned into a string : )